polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -65,6 +67,21 @@ bool isCandidate(FunctionOpInterface func) {
       !sycl::isPtrOf<sycl::NdItemType>(func.getArgumentTypes().back()))
     return false;
 
+  // Available local memory of a kernel cannot be calculated when dynamic sized
+  // local memory is used, as its size is not compile time known on device.
+  SmallVector<gpu::GPUFuncOp> kernels;
+  polygeist::getKernelCallers(func, kernels);
+  if (any_of(kernels, [](gpu::GPUFuncOp kernel) {
+        for (Value arg : kernel.getArguments())
+          if (auto memRefTy = dyn_cast<MemRefType>(arg.getType()))
```

Comment:
I assume the pointer will always be represented by a `memref` and never by an `llvm.ptr`?

Reply:
The examples I tried are all represented by a `memref`, but just in case, changed to check for `llvm.ptr` too.